model/jpa/src/main/java/org/keycloak/connections/jpa/updater/liquibase/custom/CustomCreateIndexChange.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.keycloak.connections.jpa.updater.liquibase.custom;
+
+import java.io.StringWriter;
+
+import org.jboss.logging.Logger;
+import org.keycloak.connections.jpa.updater.liquibase.LiquibaseJpaUpdaterProvider;
+import org.keycloak.connections.jpa.updater.liquibase.conn.DefaultLiquibaseConnectionProvider;
+
+import liquibase.change.AddColumnConfig;
+import liquibase.change.ChangeFactory;
+import liquibase.change.ChangeMetaData;
+import liquibase.change.ChangeParameterMetaData;
+import liquibase.change.DatabaseChange;
+import liquibase.change.core.CreateIndexChange;
+import liquibase.database.AbstractJdbcDatabase;
+import liquibase.database.Database;
+import liquibase.exception.DatabaseException;
+import liquibase.exception.UnexpectedLiquibaseException;
+import liquibase.exception.ValidationErrors;
+import liquibase.exception.Warnings;
+import liquibase.executor.ExecutorService;
+import liquibase.executor.LoggingExecutor;
+import liquibase.snapshot.InvalidExampleException;
+import liquibase.snapshot.SnapshotGeneratorFactory;
+import liquibase.sqlgenerator.SqlGeneratorFactory;
+import liquibase.statement.SqlStatement;
+import liquibase.statement.core.CreateIndexStatement;
+import liquibase.statement.core.RawSqlStatement;
+import liquibase.structure.core.Schema;
+import liquibase.structure.core.Table;
+
+/**
+ * @author <a href="mailto:yoshiyuki.tabata.jy@hitachi.com">Yoshiyuki Tabata</a>
+ */
+@DatabaseChange(name = "createIndex", description = "Creates an index on an existing column or set of columns conditionally based on the number of records.", priority = ChangeMetaData.PRIORITY_DEFAULT
+    + 1, appliesTo = "index")
+public class CustomCreateIndexChange extends CreateIndexChange {
+    private static final Logger logger = Logger.getLogger(CustomCreateIndexChange.class);
+    private int indexCreationThreshold;
+
+    @Override
+    public SqlStatement[] generateStatements(Database database) {
+        // This check is for manual migration
+        if (ExecutorService.getInstance().getExecutor(database) instanceof LoggingExecutor)
+            return super.generateStatements(database);
+
+        Object indexCreationThreshold = ((AbstractJdbcDatabase) database)
```

Comment:
What'd you say about idea to create make `DefaultLiquibaseConnectionProvider.indexCreationThreshold` `static` and to introduce 
```
    public static int getIndexCreationThreshold() {
        return indexCreationThreshold;
    }
```
there? We'd then be able to use 
```
if (DefaultLiquibaseConnectionProvider.getIndexCreationThreshold() <= 0)
                return super.generateStatements(database);
``` 
here instead of parsing `Integer` from attributes.

Reply:
I'd say if the `indexCreationThreshold` would need to be static, then the current approach is cleaner.

Reply:
@vramik @hmlnarik
I keep this implementation as-is.